stripe/src/test/java/com/stripe/android/IssuingCardPinServiceTest.kt

```diff
@@ -1,68 +1,58 @@
-import com.stripe.android.networking.ApiRequestExecutor
-import com.stripe.android.networking.ApiRequestMatcher
-import com.stripe.android.networking.StripeApiRepository
-import com.stripe.android.networking.StripeRequest
-import com.stripe.android.networking.StripeResponse
+import com.stripe.android.exception.InvalidRequestException
+import com.stripe.android.networking.AbsFakeStripeRepository
 import com.stripe.android.testharness.TestEphemeralKeyProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
 import org.json.JSONObject
 import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
 import kotlin.test.Test
 
 /**
  * Test class for [IssuingCardPinService].
  */
+@ExperimentalCoroutinesApi
 @RunWith(RobolectricTestRunner::class)
 class IssuingCardPinServiceTest {
-    private val stripeApiRequestExecutor: ApiRequestExecutor = mock()
     private val retrievalListener: IssuingCardPinService.IssuingCardPinRetrievalListener = mock()
     private val updateListener: IssuingCardPinService.IssuingCardPinUpdateListener = mock()
 
-    private val stripeRepository = StripeApiRepository(
-        ApplicationProvider.getApplicationContext(),
-        ApiKeyFixtures.FAKE_PUBLISHABLE_KEY,
-        stripeApiRequestExecutor = stripeApiRequestExecutor,
-        analyticsRequestExecutor = {}
-    )
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    private val stripeRepository = FakeStripeRepository()
     private val service = IssuingCardPinService(
         TestEphemeralKeyProvider().also {
             it.setNextRawEphemeralKey(EPHEMERAL_KEY.toString())
         },
         stripeRepository,
-        OperationIdFactory.get()
+        OperationIdFactory.get(),
+        testDispatcher
     )
 
+    @BeforeTest
+    fun setup() {
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    @AfterTest
+    fun cleanup() {
+        Dispatchers.resetMain()
+        testDispatcher.cleanupTestCoroutines()
+    }
+
     @Test
     fun testRetrieval() {
-        val response = StripeResponse(
-            200,
-            """
-            {
-                "card": "ic_abcdef",
-                "pin": "1234"
-            }
-            """.trimIndent()
-        )
-
-        whenever(
-            stripeApiRequestExecutor.execute(
-                argThat(
-                    ApiRequestMatcher(
-                        StripeRequest.Method.GET,
-                        "https://api.stripe.com/v1/issuing/cards/ic_abcdef/pin?verification%5Bone_time_code%5D=123-456&verification%5Bid%5D=iv_abcd",
-                        ApiRequest.Options("ek_test_123")
-                    )
-                )
-            )
-        ).thenReturn(response)
+        stripeRepository.retrievedPin = { "1234" }
```

Comment:
save "1234" in a val to assert in line 65, I think makes it easier to follow the code logic here